nautobot/dcim/factory.py

```diff
@@ -0,0 +1,146 @@
+import factory
+import pytz
+from factory.django import DjangoModelFactory
+
+from django.contrib.contenttypes.models import ContentType
+
+from nautobot.dcim.models import Location, LocationType, Region, Site, Device
+from nautobot.extras.models import Status
+from nautobot.ipam.models import Prefix, VLAN, VLANGroup
+from nautobot.tenancy.models import Tenant
+from nautobot.utilities.factory import random_instance, UniqueFaker
+
+
+class RegionFactory(DjangoModelFactory):
+    class Meta:
+        model = Region
+        exclude = (
+            "has_parent",
+            "has_description",
+        )
+
+    has_parent = factory.Faker("pybool")
+    parent = factory.Maybe("has_parent", random_instance(Region), None)
+    name = factory.Maybe("has_parent", UniqueFaker("city"), UniqueFaker("country"))
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+
+class SiteFactory(DjangoModelFactory):
+    class Meta:
+        model = Site
+        exclude = ("has_region", "has_tenant")
+
+    name = UniqueFaker("street_address")
+    status = random_instance(lambda: Status.objects.get_for_model(Site), allow_null=False)
+
+    has_region = factory.Faker("pybool")
+    region = factory.Maybe("has_region", random_instance(Region), None)
+
+    has_tenant = factory.Faker("pybool")
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+
+    time_zone = pytz.timezone("US/Eastern")
+    physical_address = factory.Faker("address")
+    shipping_address = factory.Faker("address")
+    latitude = factory.Faker("latitude")
+    longitude = factory.Faker("longitude")
+    contact_name = factory.Faker("name")
+    contact_phone = factory.Sequence(lambda n: f"1091-65912-{n:04d}")
+    contact_email = factory.Faker("company_email")
+
+
+class LocationTypeFactory(DjangoModelFactory):
+    class Meta:
+        model = LocationType
+        exclude = ("has_description",)
+
+    name = factory.Sequence(
+        lambda n: "Root"
+        if n == 0
+        else (
+            "Building"
+            if n == 1
+            else ("Floor" if n == 2 else ("Elevator" if n == 3 else ("Room" if n == 4 else "Aisle")))
+        )
+    )
```

Comment:
Couldn't this be something like `factory.Iterator(["Root", "Building", "Floor", "Elevator", "Room", "Aisle"])`?